create_data_image.sh

```diff
@@ -30,27 +19,27 @@ echo
 # larger so the image stays small.
 IMAGE_SIZE=$((8 + ${TOTAL_SIZE} / (920*1024)))
 
+set -e
+echo "Preparing data for the data partition..."
+rm -rf tmp-data
+mkdir tmp-data
+install -m 755 -d tmp-data/apps
+for app in apps/*; do
+	if test -f "$app"; then
+		install -m 644 -t tmp-data/apps/ "$app"
+	fi
+done
+install -m 755 -d tmp-data/local/etc/init.d
+install -m 755 resize_data_part_launcher.target-sh tmp-data/local/etc/init.d/S00resize
+install -m 755 resize_data_part.target-sh tmp-data/resize_data_part.sh
+
 echo "Creating data partition of ${IMAGE_SIZE} MB..."
 mkdir -p images
 dd if=/dev/zero of=images/data.bin bs=1M count=${IMAGE_SIZE}
 MKE2FS_CONFIG=mke2fs.conf /sbin/mkfs.ext4 -b4096 -I128 -m3 \
-		-E lazy_itable_init=0,lazy_journal_init=0,resize=268435456 \
+		-E lazy_itable_init=0,lazy_journal_init=0,resize=268435456,root_owner=0:0 \
```

Comment:
What will the owner inside the created file system be for files other than the root directory? Some files should be owned by root, while for example the pre-installed applications should be owned by the user, since the user must be allowed to remove or upgrade them.

Reply:
On my RG350, the owner of the appears to be `default` (uid `1000`).

Reply:
I guess that's because the user IDs on your PC are the same as on OpenDingux: 1000:1000. If you run the script as a different user it probably would not work. An easy fix is to chown 1000:1000 though, as this is guaranteed to be the "od" user's IDs.

Reply:
Can you please push the fix, since you have more understanding of what's going on here?
Note that this is now configured via mke2fs.conf.